tests/unit/Curl/HandlerTest.php

```diff
@@ -131,4 +131,29 @@ public function testWriteFunction()
             self::assertTrue($res);
         });
     }
+
+    /**
+     * @covers \Swoole\Curl\Handler::execute()
+     */
+    public function testResolve()
+    {
+        Coroutine\run(function () {
+            $host = 'httpbin.org';
+            $url = 'https://httpbin.org/get';
+            $ip = Coroutine::gethostbyname($host);
+            $ch = curl_init();
+
+            curl_setopt($ch, CURLOPT_URL, $url);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+            curl_setopt($ch, CURLOPT_RESOLVE, ["{$host}:443:{$ip}"]);
+
+            $data = curl_exec($ch);
+            $httpPrimaryIp = curl_getinfo($ch, CURLINFO_PRIMARY_IP);
+            curl_close($ch);
+            $body = json_decode($data, true);
+            self::assertSame($body['headers']['Host'], 'httpbin.org');
+            self::assertEquals($body['url'], $url);
+            self::assertEquals($ip, $httpPrimaryIp);
+        });
+    }
```

Comment:
I'm wondering if we should add another test case to test invalid `CURLOPT_RESOLVE ` options:

```diff
index 004bff7..45e0d83 100644
--- a/tests/unit/Curl/HandlerTest.php
+++ b/tests/unit/Curl/HandlerTest.php
@@ -156,4 +156,27 @@ class HandlerTest extends TestCase
             self::assertEquals($ip, $httpPrimaryIp);
         });
     }
+
+    /**
+     * @covers \Swoole\Curl\Handler::execute()
+     */
+    public function testInvalidResolve()
+    {
+        Coroutine\run(function () {
+            $host = 'httpbin.org';
+            $url  = 'https://httpbin.org/get';
+            $ip   = '127.0.0.1'; // An incorrect IP in use.
+
+            $ch = curl_init();
+            curl_setopt($ch, CURLOPT_URL, $url);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+            curl_setopt($ch, CURLOPT_RESOLVE, ["{$host}:443:{$ip}"]);
+            $body = curl_exec($ch);
+            $httpPrimaryIp = curl_getinfo($ch, CURLINFO_PRIMARY_IP);
+            curl_close($ch);
+
+            self::assertFalse($body);
+            self::assertSame('', $httpPrimaryIp);
+        });
+    }
 }
```